specutils/manipulation/extract_spectral_region.py

```diff
@@ -29,69 +60,36 @@ def _to_edge_pixel(subregion, spectrum):
 
     """
     # TODO: spectral regions cannot handle strictly ascending spectral axis
-    #  values. Instead, convert to length space if axis given in a desceninding
+    #  values. Instead, convert to length space if axis given in a descending
     #  unit space (e.g. frequency). We should find a more elegant solution.
     spectral_axis = spectrum.spectral_axis
+    if spectral_axis[-1] > spectral_axis[0]:
```

Comment:
I think this still leaves the limitation that the spectrum has to be *in order* even though the order now can be either way.  It's a rare, but not unheard of, case that a spectrum has overlapping or otherwise non-consecutive axes. In that case this check won't really reflect that. I wonder if we should note somewhere, maybe as a warning in the docs or similar, that spectral axes need to be monotonic?

An alternative would be to do something more complex like:
1. use `argsort` to re-sort the spectral axis
2. Do the operations we currently do on the assumed-increasing axis
3. "unsort" to get the relevant parts of the original spectrum and extract just that

But I'm not sure this is worth it if you want to get this out ASAP, so we could defer it to the future if need be.

Reply:
Hm, I think there are higher priority things that I would want to address before spending time on that case. I'm leaning toward just making an issue for follow-up, but I'll take a least a few minutes and see if something simple would work. If I don't change it to handle non-monotonic axes I'll add a note to the docs.